### PR TITLE
Scheduler(T1180373): Rollback changes made in PR #23614.

### PR DESCRIPTION
**Pull request by williamvinogradov**
Roll back by hand these changes to save types of refactoring (part of it). These changes cause a severe bug in the scheduler react version.

For the bug details see the [T1180373](https://supportcenter.devexpress.com/ticket/details/t1180373/scheduler-ui-elements-are-rendered-multiple-times-when-pressing-the-previous-next-week) ticket.

### CODE REVIEW

File: packages/devextreme/js/renovation/component_wrapper/common/template_wrapper.ts [hunk 1]
```diff
@@ -96,31 +95,50 @@ function buildTemplateContent(
...
+
+    const content = renderTemplateContent(this.props, getPublicElement($container));
+    // TODO Vinogradov: Fix the renderer function type.
+    // @ts-expect-error The renderer function's argument hasn't the full range of possible types
+    // (the Element[] type is missing).
+    replaceWith($(node), $(content));
+
+    // NOTE: This is a dispose method that called before renderTemplate.
+    return () => {
+      const $actualContainerContent = $(container).contents().toArray();
```
Comment (IlyaKhD) [inline packages/devextreme/js/renovation/component_wrapper/common/template_wrapper.ts:138]: **Minor**
```suggestion
      const $actualContainerContent = $container.contents().toArray();
```

File: packages/devextreme/js/renovation/component_wrapper/common/template_wrapper.ts [hunk 2]
```diff
@@ -96,31 +95,50 @@ function buildTemplateContent(
...
+
+    const container = node.parentNode as Element;
+    const $container = $(container);
+    const $oldContainerContent = $container.contents().toArray();
+
+    const content = renderTemplateContent(this.props, getPublicElement($container));
+    // TODO Vinogradov: Fix the renderer function type.
+    // @ts-expect-error The renderer function's argument hasn't the full range of possible types
+    // (the Element[] type is missing).
+    replaceWith($(node), $(content));
```
Comment (IlyaKhD) [inline packages/devextreme/js/renovation/component_wrapper/common/template_wrapper.ts:134]: **Doubt**
Are you sure this check is of no use?

```ts
if (content.length !== 0 && !(content.length === 1 && content[0] === container)) {
```

File: packages/devextreme/js/renovation/component_wrapper/common/__tests__/component.test.tsx [hunk 3]
```diff
@@ -981,9 +981,10 @@ describe('templates and slots', () => {
     $('#component').dxChildrenTestWidget({});
 
     const children = $('#component')[0].childNodes;
-    expect(children[2]).toBe(slotBefore[0]);
-    expect(children[3]).toBe(slotContent[0]);
-    expect(children[4]).toBe(slotAfter[0]);
+    expect(children.length).toBe(5);
+    expect(children[1]).toBe(slotBefore[0]);
```
Comment (IlyaKhD) [inline packages/devextreme/js/renovation/component_wrapper/common/__tests__/component.test.tsx:985]: **Doubt**
I'm not quite sure if the order was important.

**Review by IlyaKhD — APPROVED** [review]

File: packages/devextreme/js/renovation/component_wrapper/common/template_wrapper.ts (hunk 2 above)
Comment (williamvinogradov) [reply packages/devextreme/js/renovation/component_wrapper/common/template_wrapper.ts:134]: In the previous version, there was no this check.
So, we decided to merge it and keep an eye on it.

File: packages/devextreme/js/renovation/component_wrapper/common/__tests__/component.test.tsx (hunk 3 above)
Comment (williamvinogradov) [reply packages/devextreme/js/renovation/component_wrapper/common/__tests__/component.test.tsx:985]: It's a rollback from the previous PR.
So, it's ok

File: packages/devextreme/js/renovation/component_wrapper/common/template_wrapper.ts (hunk 1 above)
Comment (williamvinogradov) [reply packages/devextreme/js/renovation/component_wrapper/common/template_wrapper.ts:138]: I will fix it in a separate PR after this will be merged

**Review by pomahtri — APPROVED** [review]

**Pull request merged**